draft-ion.adoc

```diff
@@ -208,79 +200,176 @@ represented as follows:
+
+If a member name is not equal to the case-sensitive octet sequence `meta`, an Ion parser MUST NOT identify a member as a _Meta Object_.
+
+The member names within an identified _Meta Object_ MUST be unique; Ion parsers MUST either
+reject _Meta Object_ members with duplicate member names or use a JSON
+parser that returns only the lexically last duplicate member, as specified
 in {uri-ecmaScript-5-1-15-12}[Section 15.12 (The JSON Object)] of {uri-ecmaScript-5-1}[ECMAScript 5.1].
 
-The set of fields that a _Meta Object_ must contain to be considered valid
+The set of members that a _Meta Object_ must contain to be considered valid
 is resource-dependent.  In the absense of any resource-specific requirements,
-any field not understood by implementations MUST be ignored.
+any member not understood by implementations MUST be ignored.
+
+[[links]]
+== Links
+
+An Ion Link is an Ion Object where its `meta` member contains an {uri-IRIs}[IRI] member named `href`. An Ion Link is therefore JSON serialization of a {uri-rfc5988}[web link] to another resource.
+
+Ion parsers MUST identify any JSON structure matching this definition as an Ion Link.  Anything that does not match this defined structure MUST NOT be considered an Ion Link.
+
+A non-normative example of a minimal Ion link:
+
+.Example simplest Ion link:
+[source,json]
+----
+{ "meta": { "href": "https://api.example.com/corporations/acme" } }
+----
+
+=== Link Relation Types
+
+All Ion links have one or more <<links, link relation types>>.  A link's total set of link relation types is the set union of the link's implicit link relation type and the link's explicit link relation types.
+
+==== Implicit Link Relation Type
+
+All Ion links have exactly one implicit link relation type.  The implicit link relation type can be one of three possible values depending on where the link is located in the Ion content:
+
+1. Root Object
+2. Named Link
+3. Collection Resource `items` array element
+
+===== Root Object Implicit Link Relation Type
+
+If the Root Object is also found to be an Ion Link, it has an implicit link relation type of `self`.
+
+A non-normative example of a Root Object that is also an Ion Link:
+
+[source,json]
+----
+{
+  "meta": { "href": "https://api.example.com/users/1" },
+  "name": "Joe"
+}
+----
 
-=== Registered Metadata Field Names
+===== Named Link Implicit Link Relation Type
 
-The following field names are registered in the IANA Ion Metadata field registry
-defined in [TBD].  None of the fields defined below are intended to be mandatory
+If a JSON Object member value is an Ion Link, the Ion Link is implicitly assigned a link relation type that equals the member name.
+
+A non-normative example:
+
+[source,json]
+----
+{
+  "employer": { "meta": { "href": "https://api.example.com/corporations/acme" } }
+}
+----
+
+The above example shows an Ion Link member named `employer`.  The octet sequence `employer` is therefore assigned as the link's implicit link relation type.
```

Comment:
Good comments - I'll address inline:

> This means that (a) the server has to change how it builds the representation because it has a different name and (b) the client has to look for the order links in a different location.

I don't see how this makes a difference:

The above example shows `order` initially as a link.  Let's assume that we keep the name the same as suggested, and change the value from an object to an array as suggested.  This still requires fundamentally changing the data structure - (a) is still a concern (but instead of the name changing, the value changes).  This means the data model is not resilient to change over time - it requires both a server and client change to interpret the data correctly.

But the problem with arrays goes much deeper when talking about resiliency to change over time.  What if you wanted to represent when the orders collection was created?  The last time it was updated?  Whether or not there are other pages?  Who created collection?  etc, etc, etc.  If using raw arrays, you're stuck: you pretty much need to add additional metadata elsewhere that is less elegant and would feel clunky (e.g. an `orders` array member and a peer `ordersMetadata` object member - pretty clunky instead of representing it as a single cohesive object).

Collection Resources solve all of these problems - a collection resource can have links, primitive members, metadata, etc, and all of this can change over time and the document nesting structure and data type *never* changes.  This is the true test of a foundational mediaType that could be used to build the back of internet-scale services IMO.

In this spirit, if all collections are represented as Collection Resources, (b) is never a concern since the document structure never changes.  Even if it were a concern, there could easily be something similar to XPATH or CSS Selectors for Ion, e.g. ionDocument.findWhere("rel": ["order"]) --> (array of order links).

Given these reasons, I would definitely go so far as to say that Ion documents should *never* have any arrays other than an `items` array in a Collection Resource object.  This is the only way to guarantee a foundational structure that can change over time without breaking (m)any things.  JSON arrays can't hold metadata at all and IMO are a poor mechanism for a data interchange standard (hence the Collection Resource).

Now, whether the spec says that Ion content *MUST NOT* have arrays other than Collection Resource `items` arrays, I'm still debating.  The advantages for this requirement are so strong and the downsides practically non-existent, I'm trying to see why we wouldn't want to enforce that.  Thoughts?

Also, I should note that in my head at the moment, my intention for Ion is to be somewhat opinionated like this, to ensure cleanliness and ease of integration across services - it is not intended to work on existing previous content structures where someone could slap the media type `application/ion+json` to existing JSON content and have it 'just work'.  In Atom for example, you can't just take your existing XML collection element and expect it to 'just work' as an Atom `container` - you have to ensure your document structure adheres to the rules set forth by Atom for `container`s.  

Atom is extremely successful with this approach, as I expect Ion would be as well.